main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -1688,6 +1725,127 @@ object Lowering {
     Symbol.freshVarSym(name, BoundBy.Let, loc)
   }
 
+  private def unwrapProvenanceType(tpe: Type, loc: SourceLocation): Type = tpe match {
+    case Type.Apply(Type.Cst(TypeConstructor.Vector, _), extType, _) => extType
+    case _ => throw InternalCompilerException("Could not unwrap provenance type", loc)
+  }
+
+  private def collectPredsFromExtVarType(tpe: Type, loc: SourceLocation): List[(Name.Pred, List[Type])] = tpe match {
```

Comment:
We usually call these `fooOf` instead of `collect`. 

So this would be `predicatesOf` or similar.